component/src/test/java/org/wso2/extension/siddhi/execution/approximate/count/CountTestCase.java

```diff
@@ -292,6 +291,175 @@ public void testApproximateCount_9() throws InterruptedException {
         Assert.assertEquals(true, exceptionOccurred);
     }
 
+    @Test
+    public void testApproximateCount_10() throws InterruptedException {
+        final int windowLength = 1000;
+        final double confidence = 0.99;
+        final double relativeError = 0.01;
+
+        LOG.info("Approximate Cardinality Test Case - for Siddhi length window - " +
+                "default relative error(" + relativeError + ") and confidence(" + confidence + ") - int input");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "define stream inputStream (number int);";
+        String query = ("@info(name = 'query1') " +
+                "from inputStream#window.length(" + windowLength + ")#approximate:count(number) " +
+                "select * " +
+                "insert into outputStream;");
+
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(inStreamDefinition + query);
+
+        siddhiAppRuntime.addCallback("outputStream", new StreamCallback() {
+            @Override
+            public void receive(Event[] events) {
+                for (Event event : events) {
+                    totalEventsArrived++;
+
+                    if (totalEventsArrived < windowLength) {
+                        exactCount = (totalEventsArrived / noOfUniqueEvents) + 1;
+                    } else {
+                        exactCount = windowLength / noOfUniqueEvents;
+                    }
+
+                    lowerBound = (long) event.getData(2);
+                    upperBound = (long) event.getData(3);
+
+                    if (exactCount >= lowerBound && exactCount <= upperBound) {
+                        validEvents++;
+                    }
+                }
+                eventArrived = true;
+            }
+        });
+
+        InputHandler inputHandler = siddhiAppRuntime.getInputHandler("inputStream");
+        siddhiAppRuntime.start();
+
+        for (int j = 0; j < totalEventsSent; j++) {
+            inputHandler.send(new Object[]{j % noOfUniqueEvents});
+            Thread.sleep(1);
```

Comment:
Is it really making any difference here? You have put it 1 millisecond here. Does it make any effect on the test case? I think we can remove this. WDYT ? Please refer the other places also

Reply:
No effect. I will remove it